client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.js

```diff
@@ -11,81 +11,130 @@ import {
-  const color_scale = d3.scaleLinear()
-    .domain([0, max])
-    .range([0.2, 1]);
-
-  return max > 0 && {
-    data,
-    color_scale,
-    years: std_years,
-    formatter,
-  };
-};
+    return {
+      label: provinces[prov_code].text,
+      data: formatted_data,
+    };
+  } )
+  .filter('data')
+  .value();
 
-const prepare_data_for_a11y_table = (data) =>
-  _.chain(provinces)
-    .map((province, prov_code) => {
-      if( !_.includes(["onlessncr", "qclessncr", "ncr"], prov_code)){
-        const formatted_yearly_tp = _.map(
-          data, 
-          (row) => formats["compact2_written_raw"](row[prov_code])
-        );
-
-        return {
-          label: province.text,
-          data: formatted_yearly_tp,
-        };
-      }
-    })
-    .filter((data) => !_.isUndefined(data))
-    .value();
 
-export const declare_tp_by_region_panel = () => declare_panel({
-  panel_key: "tp_by_region",
-  levels: ["gov", "dept"],
-  panel_config_func: (level, panel_key) => ({
-    depends_on: ['orgTransferPaymentsRegion'],
-    calculate: function(subject, info, options){
-      const { orgTransferPaymentsRegion } = this.tables;
-      return calculate_common(subject, orgTransferPaymentsRegion);
-    },
-    
-    render(render_args){
-      const {
-        calculations,
-        footnotes,
-        sources,
-      } = render_args;
-      const { panel_args } = calculations;
-      const { data } = panel_args;
+class TPMap extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      show_per_capita: false,
+      loading: true,
+      population: {},
+    };
+  }
+  componentDidMount(){
+    loadPopulation()
+      .then( 
+        (transformed) => this.setState({
+          loading: false,
+          population: transformed,
+        })
+      );
+  }
+  render(){
+    const { calculations, footnotes, sources } = this.props;
+    const { show_per_capita, loading, population } = this.state;
+
+    if (loading) {
+      return (
+        <div style = {{position: "relative", height: "80px", marginBottom: "-10px"}}>
+          <SpinnerWrapper/>
+        </div>);
+    } else {
+      const { tables } = calculations.panel_args;
+      const { tp } = tables;
+
+      const changeState = () => {
+        this.setState({ show_per_capita: !show_per_capita });
+      };
+
+      const get_subject_data_for_year = (year) => tp.prov_code(
+        year,
+        calculations.subject.level === 'dept' && calculations.subject.id
+      );
+
+      //generating both data sets, as a11y table will need both later anyways
+      const tp_data = std_years.map(get_subject_data_for_year);
+      const tp_pc_data = std_years.map((year, i) => {
+        const single_year_tp_data = get_subject_data_for_year(year);
+        const result = _.chain(_.keys(single_year_tp_data))
+          .pullAll(["na", "abroad"])
+          .map((prov) => {
+            return [prov, single_year_tp_data[prov]/population[prov][i]];
+          })
```

Comment:
Body isn't necessary for this map as written, but rather than directly return `[prov, single_year_tp_data[prov]/population[prov][i]]` I think it would be better to keep the body and split this calculation in to multiple steps with clearly named variables. Maybe
```
.map((prov) => {
  const in_year_prov_transfer_payments = single_year_tp_data[prov];
  const in_year_prov_population = population[prov][i];
  return [
    prov,
    in_year_prov_transfer_payments/in_year_prov_population,
  ];
})
```

Reply:
Changed to your more explicit operations
Done